### PR TITLE
Adding ppc64le specific fixes for tests "expression/builtin_math_test.go" and "store/tikv/lock_test.go"

### PR DESCRIPTION
**Pull request by YugandhaD**
Changes made in: I have changed two test cases.

The changes were required because of the different behavior of some casting and exponential operations on ppc64le platform.  

Two tests “github.com/pingcap/tidb/store/tikv”, and “github.com/pingcap/tidb/expression”, are failing on ppc64le platform.

•	For “github.com/pingcap/tidb/store/tikv”:
This tests is failing as its subtest “lock_test.go” fails. The test fails with assertion error, which is due to the casting operation in the test where float is casted to a signed int. However, on ppc64le, all integers are unsigned integers by default, so we need to do the casting separately. I have updated it in the patch and changes I made are ppc64le specific.

•	For “github.com/pingcap/tidb/expression”:
This test case was also failing with assertion error for builtin_math_test.go. The main cause of the assertion error here was an exponential operation being performed. For math.Exp(1), both x86 and ppc64le shows up different values. For x86 the value is 2.718281828459045 and for ppc64le the value is 2.7182818284590455 where we can see a single digit precision difference. I have changed the test case accordingly in case of ppc64le.

I have raised a github issue for above test failure: https://github.com/pingcap/tidb/issues/3376

### CODE REVIEW

**Comment by ngaut** [comment]
Good job.

**Comment by shenli** [comment]
@YugandhaD CI failed. Please make sure you could pass `make check`.

**Comment by YugandhaD** [comment]
Yes, I am looking into it to fix this.
Thanks.

File: expression/builtin_math_test.go [hunk 1]
```diff
@@ -79,30 +80,58 @@ func (s *testEvaluatorSuite) TestCeil(c *C) {
...
-		{"", nil, NotNil},
-	} {
-		fc := funcs[ast.Exp]
-		f, err := fc.getFunction(datumsToConstants(types.MakeDatums(t.num)), s.ctx)
-		c.Assert(err, IsNil)
-		v, err := f.eval(nil)
-		c.Assert(err, t.err)
-		c.Assert(v, testutil.DatumEquals, types.NewDatum(t.ret))
+	if runtime.GOARCH == "ppc64le" {
+		for _, t := range []struct {
```
Comment (tiancaiamao) [inline expression/builtin_math_test.go]:
```
testcases := []struct {...} {...}
for _, t := range testcases {
}
```

File: expression/builtin_math_test.go [hunk 2]
```diff
@@ -79,30 +80,58 @@ func (s *testEvaluatorSuite) TestCeil(c *C) {
...
+		} {
+			fc := funcs[ast.Exp]
+			f, err := fc.getFunction(datumsToConstants(types.MakeDatums(t.num)), s.ctx)
+			c.Assert(err, IsNil)
+			v, err := f.eval(nil)
+			c.Assert(err, t.err)
+			c.Assert(v, testutil.DatumEquals, types.NewDatum(t.ret))
+		}
+	} else {
+		for _, t := range []struct {
```
Comment (tiancaiamao) [inline expression/builtin_math_test.go]: ditto

File: store/tikv/lock_test.go [hunk 3]
```diff
@@ -14,14 +14,14 @@
...
 import (
-	"math"
-	"time"
-
 	. "github.com/pingcap/check"
 	"github.com/pingcap/kvproto/pkg/kvrpcpb"
 	"github.com/pingcap/tidb/kv"
 	"github.com/pingcap/tidb/store/tikv/tikvrpc"
 	goctx "golang.org/x/net/context"
+	"math"
```
Comment (tiancaiamao) [inline store/tikv/lock_test.go]: Put Go standard packages first, then third-part packages, split with a newline.

**Comment by YugandhaD** [comment]
@tiancaiamao I have optimized my code as per your suggestions , and the CI for c18161d also passing now. Could you please have a look once?
Thanks.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by shenli** [comment]
LGTM
@YugandhaD Thanks!

**Comment by YugandhaD** [comment]
Thanks!

**Pull request merged**